### PR TITLE
Scaling up loading of millions of configured prefixes in db, detector, mitigation, monitor (pytricia)

### LINKED ISSUE #224: Optimize prefix expansion for millions of configured prefixes
**Describe the bug**
In the order of millions of prefixes, the prefix expansion stage may take several minutes and fill up the RAM.

**Affected Component(s)**
- [X] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

**To Reproduce**
Check with 139.0.0.0/8^+ (16 mil prefixes)

**Expected behavior**
The loading should be much faster.

**Screenshots**

### PR DESCRIPTION
**Pull request by vkotronis**
### Description of PR

What component(s) does this PR affect?

- [X] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Related Issue

Resolves #224 

### Type

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: backend/core/utils/__init__.py [hunk 1]
```diff
@@ -504,3 +488,18 @@ def ping_redis(redis_instance, timeout=5):
...
+    if prefix in pyt_tree:
+        worst_prefix = pyt_tree.get_key(prefix)
+        while pyt_tree.parent(worst_prefix):
+            worst_prefix = pyt_tree.parent(worst_prefix)
+        return worst_prefix
+    return None
+
+
+def get_ip_version(prefix):
+    if ":" in prefix:
```
Comment (vkotronis) [inline backend/core/utils/__init__.py]: TODO replace with "find" or similar, avoid scanning the entire IPv4 or IPv6 address

File: monitor/core/utils/__init__.py [hunk 2]
```diff
@@ -331,3 +310,18 @@ def ping_redis(redis_instance, timeout=5):
...
+    if prefix in pyt_tree:
+        worst_prefix = pyt_tree.get_key(prefix)
+        while pyt_tree.parent(worst_prefix):
+            worst_prefix = pyt_tree.parent(worst_prefix)
+        return worst_prefix
+    return None
+
+
+def get_ip_version(prefix):
+    if ":" in prefix:
```
Comment (vkotronis) [inline monitor/core/utils/__init__.py]: TODO replace with "find" or similar, avoid scanning the entire IPv4 or IPv6 address

Comment (vkotronis) [reply monitor/core/utils/__init__.py]: Tested with timeit: "in" is the fastest option. Resolving.

File: backend/core/utils/__init__.py (hunk 1 above)
Comment (vkotronis) [reply backend/core/utils/__init__.py]: Tested with timeit: "in" is the fastest option. Resolving.

File: backend/core/database.py [hunk 3]
```diff
@@ -666,49 +661,78 @@ def build_radix_tree(self):
...
+                                    "prefix": translated_prefix,
+                                    "data": {"confs": []},
+                                }
+                                self.prefix_tree[ip_version].insert(
+                                    translated_prefix, node
+                                )
+                            node["data"]["confs"].append(conf_obj)
+                            raw_prefix_count += 1
+                            if raw_prefix_count % 100000 == 0:
+                                log.info(
```
Comment (vkotronis) [inline backend/core/database.py]: @slowr should I used log.debug for these messages that show progress? (especially for the 100k periodic ones, I will keep log.info for the others). This applies also to detection, mitigation and monitor. What do you think?

File: backend/core/database.py [hunk 4]
```diff
@@ -666,49 +661,78 @@ def build_radix_tree(self):
...
+                            else:
+                                node = {
+                                    "prefix": translated_prefix,
+                                    "data": {"confs": []},
+                                }
+                                self.prefix_tree[ip_version].insert(
+                                    translated_prefix, node
+                                )
+                            node["data"]["confs"].append(conf_obj)
+                            raw_prefix_count += 1
```
Comment (slowr) [inline backend/core/database.py]: the +1 and also the mod might be adding delay on the operations (if we want to be picky :P)

Comment (vkotronis) [reply backend/core/database.py]: Removed %. Will keep +1 since we need the processing count for the logs (informational for the user).

**Comment by vkotronis** [comment]
> just logging and we are good!

OK I am merging then.

**Pull request merged**